callhome.go

```diff
@@ -0,0 +1,157 @@
+package netconf
+
+import (
+	"crypto/tls"
+	"fmt"
+	"github.com/nemith/netconf/transport"
+	ncssh "github.com/nemith/netconf/transport/ssh"
+	nctls "github.com/nemith/netconf/transport/tls"
+	"golang.org/x/crypto/ssh"
+	"net"
+)
+
+type TransportType uint
```

Comment:
In general i would rather see this layered where a transport is constructed and passed into the Callhome server

Reply:
Please correct me if I'm wrong, as per your implementation, creating a transport involves passing in a connection (ssh or tls or whatever depending on the desired transport). That is fine for the classical netconf connection that is initiated by the netconf-client device but callhome works the other way around (i.e. is the netconf-server initiating the connection by opening a TCP connection to the callhome server).
The transport (and the subsequent netconf connection) in case of callhome shall be created on top of the incoming TCP connection, therefore I cannot pass the transport to the callhome server, otherwise it would mean that the connection has already been established defeating the purpose of callhome. Does this make sense to you?

Reply:
100% on board.  You couldn't pass a `Transport` object in, but we could create a new interface called `CallHomeTransport` or `TransportHandler` (I am bad at naming that would take a active TCP session and create a TLS or SSH session out of it.   This way we are not adding transportation logic inside the core `netconf` package.

Reply:
I definitely agree on that. I updated the PR with a new commit that:
 - introduce the CallHomeTransport interface and implements it for TLS and SSH for end user simplicity
 - allows handling multiple clients connection

So far implementation is based on client IP address (need to wrap my mind around to understand how I can expand this nicely to manage host keys also) and it is still pretty basic.

This said, I tested with a couple of netconf devices attempting SSH callhome (one of those was configured, the other one was not), this is what you get in the output running the callhome_ssh example:
```
2023/04/12 18:55:50 callhome server listening on: 0.0.0.0:4339
handling connection from: 192.168.121.17:2386
error handling callhome connection from address: 192.168.121.17:2386, error: no client configured for address 192.168.121.17:2386
handling connection from: 192.168.121.230:3457
handling connection from: 192.168.121.17:2387
error handling callhome connection from address: 192.168.121.17:2387, error: no client configured for address 192.168.121.17:2387
2023/04/12 18:56:03 Config:
<sensitive config xml but I swear it responded correctly :) >
```

Wanted to check out with before moving on.

Maybe Callhome could be a package per se or would you rather keep it in the main netconf package?

Reply:
> Maybe Callhome could be a package per se or would you rather keep it in the main netconf package?

For me package consideration is mostly around external dependencies (thus ssh and tls being their own package).  Transport is it's own package for cylical redundancy reasons.  Adding callhome to the `netconf` package seems like the right place.  

I am not a big fan of Go modules that are a lot of different packages `netconf.CallHomeServer` is a nicer fully qualified name for an object than `callhome.Server` in my opinion.

> Any hint on if and how you want to handle logging?

Logging I am trying to hold off as long as possible to see if it is needed (it probably is).  Even then it should be optional, interface based, and off by default.  

For callhome it may be better to have handlers/callbacks for anything that a user may want to do something with (including logging)?  I haven't put a lot of thought into it yet.

Reply:
I updated the PR, I now make use of two channels in the CallHomeServer struct for the consumer to retrieve connecting CallHome clients and intercept errors, it seems a nicer way to handle it to me.
Do you have any suggestion/features you want to add to this PR?